### PR TITLE
[history_v4.9.x] Update types for history

### PR DESCRIPTION
**Pull request by micburks**
- Link to GitHub or NPM: *Covering [these functions](https://github.com/ReactTraining/history/blob/master/modules/PathUtils.js#L24-L59) from the **history** package*
- Links to documentation: *Unfortunately, these are undocumented*
- Type of contribution: *new definition | **ADDITION** | fix | refactor*

These exported APIs are undocumented and were missed in the previous libdef iteration.

Most of the diff is from prettier (`"` => `'`, `,` => `;`). eslint didn't mind, so I left them in.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ library-definition test runner.
 ***We call this the "flow version directory".***
 
 This specifies that the definition you are contributing is compatible with the
-version range of the directiry. You MUST specify a version range with names like
+version range of the directory. You MUST specify a version range with names like
```
Comment (pascalduez) [inline CONTRIBUTING.md]: Nit, but would you mind moving this change do a separate commit, or better another PR?

File: definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
-declare module "history" {
-
+// @flow
```
Comment (pascalduez) [inline definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js]: Unecessary

File: definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js [hunk 3]
```diff
@@ -92,32 +96,44 @@ declare module "history" {
...
     ) => void,
   };
 
   declare function createHashHistory(opts?: HashHistoryOpts): HashHistory;
+
+  declare type LocationType = {
+    pathname?: string,
+    search?: string,
+    hash?: string,
+  };
```
Comment (pascalduez) [inline definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js]: There's already `BrowserLocation`, `MemoryLocation`, `HashLocation` (which should be discussed BTW...)   
Why creating another one?

File: CONTRIBUTING.md (hunk 1 above)
Comment (micburks) [reply CONTRIBUTING.md]: 👍

File: definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js (hunk 3 above)
Comment (micburks) [reply definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js]: Yeah I see what you mean here. I can remove this type and use a union of all Location types in the following 2 functions

File: definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js (hunk 2 above)
Comment (micburks) [reply definitions/npm/history_v4.9.x/flow_v0.25.x-/history_v4.9.x.js]: Oops 😑
